Model/Client.php

```diff
@@ -135,4 +135,28 @@ public function getAllowedGrantTypes()
     {
         return $this->allowedGrantTypes;
     }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function eraseCredentials(): void
+    {
+        // nothind to erase
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getRoles(): array
+    {
+        return ['ROLE_USER'];
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getUserIdentifier(): string
+    {
+        return $this->getRandomId();
+    }
```

Comment:
A `Client` is not the same thing as a user, which is entirely up to the application using this package to implement.  What need did adding these methods seem to serve?  Perhaps there's more to a fix here than just removing these.

Reply:
https://github.com/FriendsOfSymfony/oauth2-php/commit/b59f92b963c0214ba505abedaef6a8c177465e87

The oauth2-php fork I was using added implements UserInterface to the Client model that this was extending so it was throwing a fatal error to not have those functions.  I'm not sure what purpose adding that had, but all the tests pass without it, so I removed it.